isbg/isbg.py

```diff
@@ -1,909 +1,613 @@
-    --savepw             Store the password to be used in future runs
-    --spamc              Use spamc instead of standalone SpamAssassin binary
-    --spaminbox mbox     Name of your spam folder
-    --nossl              Don't use SSL to connect to the IMAP server
-    --teachonly          Don't search spam, just learn from folders
-    --trackfile file     Override the trackfile name
-    --verbose            Show IMAP stuff happening
-    --verbose-mails      Show mail bodies (extra-verbose)
-    --version            Show the version information
-
-    (Your inbox will remain untouched unless you specify --flag or --delete)
 
-"""
+For any entries that match, the message is copied to another folder, and the
+original marked or deleted.
 
-import sys  # Because sys.stderr.write() is called bellow
-from io import BytesIO
+.. autodata:: __version__
 
-# FIXME: This is necessary to allow using isbg both straight from the repo and installed / as an import.
-# We should probably decide to not care about running isbg as top-level script straight from the repo.
-try:
-    from .sa_unwrap import unwrap
-except:
-    try:
-        from sa_unwrap import unwrap
-    except ImportError:
-        sys.stderr.write('Cannot load sa_unwrap, please install isbg package properly!\n')
-        # Create No-Op dummy function
-        unwrap = lambda x: None
+.. autodata:: __exitcodes__
+    :annotation: = {'ok': 0, ...}
 
-try:
-    from docopt import docopt  # Creating command-line interface
-except ImportError:
-    sys.stderr.write("Missing dependency: docopt\n")
-    raise
+"""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
 
-from subprocess import Popen, PIPE
 
-import imaplib
-import re
 import os
-import getpass
-import string
-import time
+import sys     # Because sys.stderr.write() is called bellow
+
+from isbg import imaputils
+from isbg import secrets
+from isbg import spamproc
+from isbg import utils
+
+from .utils import __
+
 import atexit
+import getpass
 import json
 import logging
+import re
+import time
 
+# xdg base dir specification (only xdg_cache_home is used)
 try:
-    from hashlib import md5
+    from xdg.BaseDirectory import xdg_cache_home
 except ImportError:
-    from md5 import md5
+    xdg_cache_home = ""    # pylint: disable=invalid-name
+if xdg_cache_home == "":
+    # pylint: disable=invalid-name
+    xdg_cache_home = os.path.expanduser("~" + os.sep + ".cache")
+    """str: From the `XDG Base Directory specification`_.
+
+We used this directory to create a `isbg/` one to store cached data:
+    * lock file.
+    * password file.
+    * chached lists of ``uids``.
+
+Note:
+    For best `xdg_cache_home` detection, install *python-xdg*.
+
+.. _XDG Base Directory specification:
+    https://standards.freedesktop.org/basedir-spec/basedir-spec-latest.html
+"""
+
+
+__version__ = "2.0.0"  #: The current isbg version
```

Comment:
I plan to merge this a `2.1.0`. If that's alright with you, changing it now would be nice.